packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -223,3 +243,31 @@ const appendDevBrowserOnDevOrStaging = (req: NextRequest, res: Response) => {
   }
   return res;
 };
+
+// - Default behavior:
+//    If the route path is `['/api/(.*)*', '*/trpc/(.*)']`
+//    or Request has `Content-Type: application/json`
+//    or Request method is not-GET,OPTIONS,HEAD,
+//    then this is considered an API route.
+//
+// - If the user has provided a specific `apiRoutes` prop in `authMiddleware` then all the above are discarded,
+//   and only routes that match the user’s provided paths are considered API routes.
+const createApiRoutes = (apiRoutes: RouteMatcherParam | undefined): ((req: NextRequest) => boolean) => {
+  if (apiRoutes) {
+    return createRouteMatcher(apiRoutes);
+  } else {
+    const isDefaultApiRoute = createRouteMatcher(DEFAULT_API_ROUTES);
+    return (req: NextRequest) =>
+      isDefaultApiRoute(req) || isRequestMethodIndicatingApiRoute(req) || isRequestContentTypeJson(req);
+  }
+};
+
+const isRequestContentTypeJson = (req: NextRequest): boolean => {
+  const requestContentType = req.headers.get(constants.Headers.ContentType);
+  return requestContentType === constants.ContentTypes.Json;
+};
+
+const isRequestMethodIndicatingApiRoute = (req: NextRequest): boolean => {
+  const requestMethod = req.method.toLowerCase();
+  return requestMethod !== 'get' && requestMethod !== 'head' && requestMethod !== 'options';
```

Comment:
🔧 i don't think we should include `get` hear. There may be an API route that we want to retrieve data with `get`.

Reply:
@dimkl This is just an indication for our heuristic algorithm. If it's not a `GET` we are sure it's an API route, if it's a `GET` we don't know if this is a page or API endpoint, so it should return `false`.